### ISSUE TITLE
Helper function abstracts over module parameters, with sometimes disastrous consequences for printing

### ISSUE THREAD

**Issue by andreasabel** [agda/agda#2271]
Helper function now abstracts over module parameters, with disastrous consequences for printing

**Comment by UlfNorell** [comment]
Do you have an example?

**Comment by andreasabel** [comment]
> Do you have an example?

Yes.

``` agda
record Fun : Set₁ where
   field
     _⇒_ : Set → Set → Set

module Param (fun : Fun) (open Fun fun) where

   test : ∀{A} → A ⇒ A
   test = {!aux!} -- C-c C-h
-- EXPECTED:  aux : ∀ {A} → A ⇒ A
-- ACTUALLY:  aux : ∀ {fun A} → (fun Fun.⇒ A) A
```

This might not be a regression though, but still bad enough to make C-c C-h unusable in practice.

**Comment by andreasabel** [comment]
On 21.10.2016 07:10, Ulf Norell wrote:
> Do you have an example?

Yes.

```agda
record Fun : Set₁ where
    field
      _⇒_ : Set → Set → Set

module Param (fun : Fun) (open Fun fun) where

    test : ∀{A} → A ⇒ A
    test = {!aux!} -- C-c C-h
-- EXPECTED:  aux : ∀ {A} → A ⇒ A
-- ACTUALLY:  aux : ∀ {fun A} → (fun Fun.⇒ A) A
```

This might not be a regression though, but still bad enough to make C-c 
C-h unusable in practice.

-- 
Andreas Abel  <><      Du bist der geliebte Mensch.

Department of Computer Science and Engineering
Chalmers and Gothenburg University, Sweden

andreas.abel@gu.se
http://www2.tcs.ifi.lmu.de/~abel/

**Comment by andreasabel** [comment]
On 21.10.2016 07:10, Ulf Norell wrote:
> Do you have an example?

Yes.

```agda
record Fun : Set₁ where
   field
     _⇒_ : Set → Set → Set

module Param (fun : Fun) (open Fun fun) where

   test : ∀{A} → A ⇒ A
   test = {!aux!} -- C-c C-h
-- EXPECTED:  aux : ∀ {A} → A ⇒ A
-- ACTUALLY:  aux : ∀ {fun A} → (fun Fun.⇒ A) A
```

This might not be a regression though, but still bad enough to make C-c 
C-h unusable in practice.

-- 
Andreas Abel  <><      Du bist der geliebte Mensch.

Department of Computer Science and Engineering
Chalmers and Gothenburg University, Sweden

andreas.abel@gu.se
http://www2.tcs.ifi.lmu.de/~abel/

**Comment by andreasabel** [comment]
It seems better that the helper-function does the minimal abstraction possible.  Concretely, only abstract over the things needed to type the with-arguments.  In terms of `splitTelForWith`, do not abstract over `delta2`.

**Comment by andreasabel** [comment]
The relevant code is here: https://github.com/agda/agda/blob/8da239b975a76c8edf735640ae4b0237aa4fca0a/src/full/Agda/Interaction/BasicOps.hs#L866-L867
Here are some entry point concerning module parameters:
https://github.com/agda/agda/blob/8da239b975a76c8edf735640ae4b0237aa4fca0a/src/full/Agda/TypeChecking/Monad/Signature.hs#L976-L978
https://github.com/agda/agda/blob/8da239b975a76c8edf735640ae4b0237aa4fca0a/src/full/Agda/TypeChecking/Monad/Signature.hs#L989-L997

**Comment by andreasabel** [comment]
This was fixed by #6222.
